BB-67 migrate to github actions

tcarmet opened the pull request.

tcarmet committed:
BB-67 migration to GitHub Actions

bert-e commented:
# Hello tcarmet,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/backbeat/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-67 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `7.10.1`

* `7.4.10`

* `8.2.9`

* `8.3.0`


Please check the `Fix Version/s` of BB-67, or the target
branch of this pull request.

jonathan-gramain committed:
BB-67 add ADVERTISED_HOST/ADVERTISED_PORT env to kafka container

This is required to advertise the proper hostname ("localhost") to the
kafka producer/consumer, without it producers are not able to receive
the delivery reports properly as the default hostname is not
reachable.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/BB-67-migrate-gh-actions` into
`development/7.4`
* [w/7.10/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/7.10...w/7.10/feature/BB-67-migrate-gh-actions) will be merged into `development/7.10` (pull request #1995)
* [w/8.2/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/feature/BB-67-migrate-gh-actions) will be merged into `development/8.2` (pull request #1996)
* [w/8.3/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/8.3...w/8.3/feature/BB-67-migrate-gh-actions) will be merged into `development/8.3` (pull request #1997)





*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

bert-e commented:
# History mismatch



Merge commit #5082c8cd42c50ffe28ec6fb5372448023b16fb2f on the integration branch
`w/8.3/feature/BB-67-migrate-gh-actions` is merging a branch which is neither the current
branch `feature/BB-67-migrate-gh-actions` nor the development branch
`development/8.3`.

It is likely due to a rebase of the branch `feature/BB-67-migrate-gh-actions` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# History mismatch



Merge commit #3ed0b052adcd5d0f47d475b75c753ed07c916c83 on the integration branch
`w/8.3/feature/BB-67-migrate-gh-actions` is merging a branch which is neither the current
branch `feature/BB-67-migrate-gh-actions` nor the development branch
`development/8.3`.

It is likely due to a rebase of the branch `feature/BB-67-migrate-gh-actions` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/BB-67-migrate-gh-actions` into
`development/7.4`
* [w/7.10/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/7.10...w/7.10/feature/BB-67-migrate-gh-actions) will be merged into `development/7.10` (pull request #1995)
* [w/8.2/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/feature/BB-67-migrate-gh-actions) will be merged into `development/8.2` (pull request #1996)
* [w/8.3/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/8.3...w/8.3/feature/BB-67-migrate-gh-actions) will be merged into `development/8.3` (pull request #1998)





*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

miniscruff commented on .github/workflows/tests.yaml:
```diff
@@ -0,0 +1,115 @@
+    - name: Run unit tests
+      run: yarn test
+    - name: Run bucket scanner unit tests
+      run: ginkgo -r --randomizeAllSpecs --randomizeSuites --failOnPending --cover --trace --race --progress -nodes 1 -stream -timeout 5m -slowSpecThreshold 60
```
small thing here that we can change later, the bucket scanner can be a separate workflow as it is the only part using go and ginkgo

tcarmet replied:
ooh ok, yes indeed. I added here because I was mostly following what was being done in Eve.

I'll let you track this one and add it as soon as this PR is merged.

miniscruff replied:
Made https://scality.atlassian.net/browse/BB-78

bert-e commented:
# History mismatch



Merge commit #19b9ce2ee9427d04cee5de8ceb1830f89048f65f on the integration branch
`w/8.3/feature/BB-67-migrate-gh-actions` is merging a branch which is neither the current
branch `feature/BB-67-migrate-gh-actions` nor the development branch
`development/8.3`.

It is likely due to a rebase of the branch `feature/BB-67-migrate-gh-actions` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/BB-67-migrate-gh-actions` into
`development/7.4`
* [w/7.10/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/7.10...w/7.10/feature/BB-67-migrate-gh-actions) will be merged into `development/7.10` (pull request #1995)
* [w/8.2/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/feature/BB-67-migrate-gh-actions) will be merged into `development/8.2` (pull request #1996)
* [w/8.3/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/8.3...w/8.3/feature/BB-67-migrate-gh-actions) will be merged into `development/8.3` (pull request #1999)





*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/BB-67-migrate-gh-actions` into
`development/7.4`
* [w/7.10/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/7.10...w/7.10/feature/BB-67-migrate-gh-actions) will be merged into `development/7.10` (pull request #1995)
* [w/8.2/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/feature/BB-67-migrate-gh-actions) will be merged into `development/8.2` (pull request #1996)
* [w/8.3/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/8.3...w/8.3/feature/BB-67-migrate-gh-actions) will be merged into `development/8.3` (pull request #2000)





*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

bert-e commented:
# History mismatch



Merge commit #c8f40c372eb0a1d93069388b24c5da5ea8a8ad9f on the integration branch
`w/8.3/feature/BB-67-migrate-gh-actions` is merging a branch which is neither the current
branch `feature/BB-67-migrate-gh-actions` nor the development branch
`development/8.3`.

It is likely due to a rebase of the branch `feature/BB-67-migrate-gh-actions` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/BB-67-migrate-gh-actions` into
`development/7.4`
* [w/7.10/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/7.10...w/7.10/feature/BB-67-migrate-gh-actions) will be merged into `development/7.10` (pull request #1995)
* [w/8.2/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/feature/BB-67-migrate-gh-actions) will be merged into `development/8.2` (pull request #1996)
* [w/8.3/feature/BB-67-migrate-gh-actions](https://github.com/scality/backbeat/compare/development/8.3...w/8.3/feature/BB-67-migrate-gh-actions) will be merged into `development/8.3` (pull request #2005)





*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

tcarmet commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/8.2`

* :heavy_check_mark: `development/8.3`




There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/8.2`

* :heavy_check_mark: `development/8.3`




Please check the status of the associated issue BB-67.

Goodbye tcarmet.
